From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-add-basic-tests-for-lazy_mmu-fix
Date: Tue Dec 16 08:07:21 PM PST 2025

export ppc64_tlb_batch and __flush_tlb_pending to modules for
mm/tests/lazy_mmu_mode_kunit.ko.

Cc: Kevin Brodsky <kevin.brodsky@arm.com>
Cc: Ritesh Harjani (IBM) <ritesh.list@gmail.com>
From: Ritesh Harjani (IBM) <ritesh.list@gmail.com>
Subject: mm-add-basic-tests-for-lazy_mmu-fix-fix
Date: Wed, 17 Dec 2025 21:16:14 +0530

use EXPORT_SYMBOL_IF_KUNIT()

Link: https://lkml.kernel.org/r/87a4zhkt6h.ritesh.list@gmail.com
Signed-off-by: Ritesh Harjani (IBM) <ritesh.list@gmail.com>
Cc: Kevin Brodsky <kevin.brodsky@arm.com>
From: Kevin Brodsky <kevin.brodsky@arm.com>
Subject: mm-add-basic-tests-for-lazy_mmu-fix-fix-fix
Date: Wed, 17 Dec 2025 16:38:12 +0000

move MODULE_IMPORT_NS(), add comment

Link: https://lkml.kernel.org/r/20251217163812.2633648-2-kevin.brodsky@arm.com
Signed-off-by: Kevin Brodsky <kevin.brodsky@arm.com>
Cc: Ritesh Harjani (IBM) <ritesh.list@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/powerpc/mm/book3s64/hash_tlb.c |    4 +++-
 mm/tests/lazy_mmu_mode_kunit.c      |    3 +++
 2 files changed, 6 insertions(+), 1 deletion(-)

--- a/arch/powerpc/mm/book3s64/hash_tlb.c~mm-add-basic-tests-for-lazy_mmu-fix
+++ a/arch/powerpc/mm/book3s64/hash_tlb.c
@@ -25,11 +25,12 @@
 #include <asm/tlb.h>
 #include <asm/bug.h>
 #include <asm/pte-walk.h>
-
+#include <kunit/visibility.h>
 
 #include <trace/events/thp.h>
 
 DEFINE_PER_CPU(struct ppc64_tlb_batch, ppc64_tlb_batch);
+EXPORT_SYMBOL_IF_KUNIT(ppc64_tlb_batch);
 
 /*
  * A linux PTE was changed and the corresponding hash table entry
@@ -154,6 +155,7 @@ void __flush_tlb_pending(struct ppc64_tl
 		flush_hash_range(i, local);
 	batch->index = 0;
 }
+EXPORT_SYMBOL_IF_KUNIT(__flush_tlb_pending);
 
 void hash__tlb_flush(struct mmu_gather *tlb)
 {
--- a/mm/tests/lazy_mmu_mode_kunit.c~mm-add-basic-tests-for-lazy_mmu-fix
+++ a/mm/tests/lazy_mmu_mode_kunit.c
@@ -2,6 +2,9 @@
 #include <kunit/test.h>
 #include <linux/pgtable.h>
 
+/* For some symbols referenced by arch_{enter,leave}_lazy_mmu_mode on powerpc */
+MODULE_IMPORT_NS("EXPORTED_FOR_KUNIT_TESTING");
+
 static void expect_not_active(struct kunit *test)
 {
 	KUNIT_EXPECT_FALSE(test, is_lazy_mmu_mode_active());
_
